lib/guardian.rb

```diff
@@ -323,6 +323,15 @@ def can_use_primary_group?(user, group_id = nil)
     (group ? !group.automatic : false)
   end
 
+  def can_use_flair_group?(user, group_id = nil)
+    return false if !user || !group_id || !user.group_ids.include?(group_id.to_i)
+
+    group = Group.find_by(id: group_id.to_i)
```

Comment:
Minor but instead of creating an AR object here, can we instead only pluck the required columns?